invoiceregistry.go

```diff
@@ -679,7 +679,7 @@ func (i *InvoiceRegistry) process(ctx context.Context, h *registryHtlc) error {
 	if statelessData.preimage.Hash() != h.rHash {
 		logger.Debugw("Hash mismatch")
 
-		h.resolve(NewFailResolution(ResultInvoiceNotFound))
+		h.resolve(NewFailResolution(ResultAddressMismatch))
```

Comment:
The commit message lists a set of changes, which is already an indication that they probably could have been separate commits. I think three:
1. Fix return code
2. Refactor tests / add util func
3. Add more mpp coverage